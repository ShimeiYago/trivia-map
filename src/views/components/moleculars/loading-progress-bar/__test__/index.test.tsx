import { shallow, ShallowWrapper } from 'enzyme';
import { LoadingProgressBar, State, Props } from '..';

let shallowWrapper: ShallowWrapper<Props, State, LoadingProgressBar>;

const props: Props = {
  loadedPages: 1,
  totalPages: 5,
  fetchingState: 'waiting',
  isMobile: false,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<LoadingProgressBar {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('on loading', () => {
    shallowWrapper.setProps({ fetchingState: 'loading' });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('init of loading', () => {
    shallowWrapper.setProps({
      totalPages: undefined,
      fetchingState: 'loading',
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('mobile view', () => {
    shallowWrapper.setProps({
      isMobile: true,
    });
    shallowWrapper.setState({
      showLoadingProgressBar: true,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<LoadingProgressBar {...props} />);
  });

  it('should show loading progress bar if markersLoading changes to loading', () => {
    const prevProps: Props = props;
    shallowWrapper.setProps({ fetchingState: 'loading' });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate(prevProps);
    expect(instance.state.showLoadingProgressBar).toBeTruthy();
  });

  it('do not close loading progress bar soon even if markersLoading changes from loading', () => {
    const prevProps: Props = { ...props, fetchingState: 'loading' };
    shallowWrapper.setProps({ fetchingState: 'success' });
    shallowWrapper.setState({ showLoadingProgressBar: true });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate(prevProps);
    expect(instance.state.showLoadingProgressBar).toBeTruthy();
  });
});

describe('closeLoadingProgressBar', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<LoadingProgressBar {...props} />);
  });

  it('should close loading progress bar', async () => {
    shallowWrapper.setState({ showLoadingProgressBar: true });
    const instance = shallowWrapper.instance();
    instance.closeTimeMs = 0;

    await instance['closeLoadingProgressBar']();
    expect(instance.state.showLoadingProgressBar).toBeFalsy();
  });
});
