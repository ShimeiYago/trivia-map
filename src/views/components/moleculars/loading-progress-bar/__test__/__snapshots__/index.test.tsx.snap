// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `""`;

exports[`Shallow Snapshot Tests init of loading 1`] = `
<DialogScreen
  position="bottom"
  theme="white"
>
  <ForwardRef(Box)
    sx={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "width": "30px",
        }
      }
    >
      <ForwardRef(CircularProgress)
        size={20}
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)>
      <ForwardRef(Typography)
        color="text.primary"
        variant="body1"
      >
        データを読み込んでいます...
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "mr": 1,
          "width": "100%",
        }
      }
    >
      <ForwardRef(LinearProgress)
        color="primary"
        value={0}
        variant="determinate"
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)
      sx={
        Object {
          "minWidth": 35,
        }
      }
    >
      <ForwardRef(Typography)
        color="text.secondary"
        variant="body2"
      >
        0%
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</DialogScreen>
`;

exports[`Shallow Snapshot Tests mobile view 1`] = `
<DialogScreen
  position={75}
  theme="white"
>
  <ForwardRef(Box)
    sx={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "width": "30px",
        }
      }
    >
      <ForwardRef(CircularProgress)
        size={20}
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)>
      <ForwardRef(Typography)
        color="text.primary"
        variant="body1"
      >
        データを読み込んでいます...
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "mr": 1,
          "width": "100%",
        }
      }
    >
      <ForwardRef(LinearProgress)
        color="primary"
        value={20}
        variant="determinate"
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)
      sx={
        Object {
          "minWidth": 35,
        }
      }
    >
      <ForwardRef(Typography)
        color="text.secondary"
        variant="body2"
      >
        20%
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</DialogScreen>
`;

exports[`Shallow Snapshot Tests on loading 1`] = `
<DialogScreen
  position="bottom"
  theme="white"
>
  <ForwardRef(Box)
    sx={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "width": "30px",
        }
      }
    >
      <ForwardRef(CircularProgress)
        size={20}
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)>
      <ForwardRef(Typography)
        color="text.primary"
        variant="body1"
      >
        データを読み込んでいます...
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "mr": 1,
          "width": "100%",
        }
      }
    >
      <ForwardRef(LinearProgress)
        color="primary"
        value={20}
        variant="determinate"
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)
      sx={
        Object {
          "minWidth": 35,
        }
      }
    >
      <ForwardRef(Typography)
        color="text.secondary"
        variant="body2"
      >
        20%
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</DialogScreen>
`;
